### PR TITLE
[MRG] windows directories

### PR DESCRIPTION
**Pull request by untzag**
fix for uploading subdirectories from windows

### CODE REVIEW

**Comment by untzag** [comment]
broken tests...

I will come back to this

**Comment by untzag** [comment]
I believe that the `test_recursive_upload_with_subdir` test is faulty

with the changes I made in `cli.py`, redundant separators are passed into `store.create_file` if source and root are the same directory

this is just fine because the first thing you do within `create_file` is call `norm_remote_path`, which [removes any redundant separators](https://docs.python.org/2/library/os.path.html#os.path.normpath)

but since you test based on what goes _into_ `create_file`, the redundant separators break your tests (see my first commit)

to fix this, I have added in an additional call to `os.path.normpath` immediately prior to `create_file`---this makes the tests pass again

I believe the tests should be updated, but I don't want to start messing with existing tests without maintainer approval

File: osfclient/cli.py [hunk 1]
```diff
@@ -266,19 +266,14 @@ def upload(args):
...
 
         for root, _, files in os.walk(args.source):
-            # these are extra subdirectories we have walked into since the root
-            # directory, have to clean off leading slashes from their name
-            # for path.join() to work later on
-            subdir_path = root.replace(args.source, '')
-            if subdir_path.startswith('/'):
-                subdir_path = subdir_path[1:]
-
+            subdir_path = os.path.relpath(root, args.source)
```
Comment (betatim) [inline osfclient/cli.py]: Nice! I didn't know this was a thing :)

**Comment by betatim** [comment]
Updating the test sounds like a good plan to me.

Also, sorry for the slow replies.

File: osfclient/cli.py (hunk 1 above)
Comment (untzag) [reply osfclient/cli.py]: 😘

**Comment by untzag** [comment]
@betatim OK I'll probably look at tests this weekend

I guess some windows-style paths should also be tested against...

**Comment by untzag** [comment]
@betatim this PR is ready as far as I'm concerned

I decided to do the lazy thing and just directly make the tests work

you might consider adding a windows service like [appveyor](https://www.appveyor.com/) to your testing suite

**Comment by betatim** [comment]
Merged!

Having appveyor would be cool. As you can tell I am a total know-nothing-about-Windows, so if you know how to setup appveyor or create some of the scripts needed: do it :) Without access to windows box for debugging this is on my list of "big tasks I need a day for".

---

Hi @untzag  we operate under what is often called "the pull request hack" (see #10). Basically once some of your work has been merged we add you to the list of people who can merge other's work. Would that be Ok with you?

**Comment by untzag** [comment]
yeah I'm happy to be a PR reviewer

**Comment by betatim** [comment]
Welcome aboard!

**Comment by untzag** [comment]
@betatim how does it work if I want to make a PR?

shall I fork or branch inside of this repo?

**Comment by betatim** [comment]
fork and branch in your repo. Just like you did for this PR. This way it is the same for everyone who is contributing, no matter if they can write to this repo or not.

**Comment by betatim** [comment]
Should we make this more explicit in https://github.com/dib-lab/osf-cli/blob/master/CONTRIBUTING.md?

**Comment by untzag** [comment]
thanks for the clarification @betatim 

yeah contributing could be clearer---will submit a PR with small change

**Pull request merged**